catalog/files/impl/src/main/java/org/projectnessie/catalog/files/secrets/SecretsHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.files.secrets;
+
+import static java.util.Collections.emptyMap;
+
+import jakarta.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import org.projectnessie.nessie.immutables.NessieImmutable;
+
+public final class SecretsHelper {
+  private SecretsHelper() {}
+
+  /**
+   * Helper to resolve the secrets described via {@link Specializeable}s using the given
+   * secrets-provider.
+   *
+   * <p>Individual secrets are resolved in the following order:
+   *
+   * <ol>
+   *   <li>The value of the attribute on the {@code specific} object,
+   *   <li>The resolved secret for the specific object (lookup via {@code baseName + '.' +
+   *       specificName + '.' + Specializeable.name()},
+   *   <li>The value of the attribute on the {@code base} object,
+   *   <li>The resolved secret for the base object (lookup via {@code baseName + '.' +
+   *       Specializeable.name()},
+   * </ol>
+   */
+  public static <R, B> B resolveSecrets(
```

Comment:
Hard to follow, even with the javadocs above. Looks slightly over-engineered, and I wonder if this will play nicely with Quarkus own secrets management: will this work with the Quarkus Vault extension? Or with the Quarkus Kubernetes Config extension?

Reply:
Also: the return value seems ignored everywhere.